analytics/views.py

```diff
@@ -1132,6 +1134,21 @@ def support(request: HttpRequest) -> HttpResponse:
             attach_discount_to_realm(realm, new_discount)
             msg = f"Discount of {realm.string_id} changed to {new_discount} from {current_discount} "
             context["success_message"] = msg
+        elif request.POST.get("new_subdomain", None) is not None:
+            new_subdomain = request.POST.get("new_subdomain")
+            old_subdomain = realm.string_id
+            try:
+                get_realm(new_subdomain)
+            except Realm.DoesNotExist:
```

Comment:
Actually use `zerver.forms.check_subdomain_available` for doing the validation. Just checking whether a realm exists is not good enough. Sorry for not catching it before.